### PR TITLE
lie algebra and skew-sym

### PR DESCRIPTION
**Pull request by nguigs**
Add is_skew_sym method and corresponding tests

### CODE REVIEW

File: geomstats/backend/tensorflow/__init__.py [hunk 1]
```diff
@@ -31,6 +31,10 @@ def get_mask_i_float(i, n):
     return mask_i_float
 
 
+def get_from_sparse(indices, data, shape):
```
Comment (ninamiolane) [inline geomstats/backend/tensorflow/__init__.py:34]: is `get_from_sparse` the name of the numpy function?

If not, does numpy have a similar function and could we use its name and syntax?

File: geomstats/geometry/lie_algebra.py [hunk 2]
```diff
@@ -21,7 +21,7 @@ def __init__(self, dimension, n):
         dimension: int
             The dimension of the Lie algebra as a real vector space
         n: int
-            The amount of rows and columns in the matrx representation of the
+            The amount of rows and columns in the matrix representation of the
             Lie algebra
```
Comment (ninamiolane) [inline geomstats/geometry/lie_algebra.py:25]: "." missing at the end of the parameter descriptions.

(Sorry, I know it's not your code, thanks for cleaning it already!)

File: geomstats/geometry/skew_symmetric_matrices.py [hunk 3]
```diff
@@ -25,11 +30,24 @@ def __init__(self, n):
...
+
+        else:
+            for row in gs.arange(n - 1):
+                for col in gs.arange(row + 1, n):
+                    self.basis[loop_index, row, col] = 1
+                    self.basis[loop_index, col, row] = -1
+                    loop_index += 1
+
+    def belongs(self, mat, atol=TOLERANCE):
+        """Check if mat belongs to the vector space of symmetric matrices."""
```
Comment (ninamiolane) [inline geomstats/geometry/skew_symmetric_matrices.py:49]: Parameter descriptions?

File: tests/test_lie_group.py [hunk 4]
```diff
@@ -1,6 +1,4 @@
-"""
-Unit tests for Lie groups.
-"""
+"""Unit tests for Lie groups."""
```
Comment (ninamiolane) [inline tests/test_lie_group.py:1]: Thanks!

File: geomstats/geometry/skew_symmetric_matrices.py [hunk 5]
```diff
@@ -25,11 +30,24 @@ def __init__(self, n):
         self.basis = gs.zeros((dimension, n, n))
         loop_index = 0
 
-        for row in gs.arange(n - 1):
-            for col in gs.arange(row + 1, n):
-                self.basis[loop_index, row, col] = 1
-                self.basis[loop_index, col, row] = -1
-                loop_index += 1
+        if tests.tf_backend():
```
Comment (ninamiolane) [inline geomstats/geometry/skew_symmetric_matrices.py:33]: - Maybe create a method: `canonical_basis`?

- We should also avoid testing for backend in the code (`tf_backend()`): can't we abstract this by calling only backend functions?

File: geomstats/backend/tensorflow/__init__.py (hunk 1 above)
Comment (nguigs) [reply geomstats/backend/tensorflow/__init__.py:34]: - `get_from_sparse` is not the numpy of a numpy function
- I don't know any similar function per se, there a different ways of doing it in numpy e.g. fromfunction. My idea here would be to add this function to all backends using each specific tools, just like `to_ndarray` does

File: geomstats/geometry/skew_symmetric_matrices.py (hunk 5 above)
Comment (nguigs) [reply geomstats/geometry/skew_symmetric_matrices.py:33]: in the long term yes, maybe once we have a `get_from_sparse` for all backends, or stg similar with assignements

File: geomstats/backend/tensorflow/__init__.py (hunk 1 above)
Comment (ninamiolane) [reply geomstats/backend/tensorflow/__init__.py:34]: OK! Maybe call it `sparse_array()`? Sorry I'm nit-picking, but get_from_sparse is not super intuitive?

Comment (nguigs) [reply geomstats/backend/tensorflow/__init__.py:34]: Yes but it's not a sparse array in the usual sense, but a dense one. Sparse refers to scipy.sparse or SparseTensor which are different data types that don't store all the zeros. The proposed method converts a sparse array to a dense (usual) one.

Comment (nguigs) [reply geomstats/backend/tensorflow/__init__.py:34]: Maybe `array_from_sparse`?

Comment (ninamiolane) [reply geomstats/backend/tensorflow/__init__.py]: Got it! `array_from_sparse` would be great: the naming is also closer to our conversion functions.

**Pull request merged**